db/flush_job.cc

```diff
@@ -306,6 +317,272 @@ void FlushJob::Cancel() {
   base_->Unref();
 }
 
+Status FlushJob::MemPurge(autovector<MemTable*>& purged_mems) {
+  Status s;
+  db_mutex_->AssertHeld();
+  db_mutex_->Unlock();
+  assert(!mems_.empty());
+
+  // Store the full output memtables in
+  // autovector "purged_mems".
+  // autovector<MemTable*> purged_mems = {};
+  purged_mems = {};
+
+  MemTable* new_mem = nullptr;
```

Comment:
Prefer `unique_ptr` to naked pointer when reasonably possible

Reply:
Agreed. I tried and this is just compliacted here, because you inherently need to lose ownership of the pointer to non-unique_ptr compatible objects. E.g: when adding the imm(), which only accepts raw pointers, we need to push back the pointer and not free the memory.

Reply:
Can use `.release()`